dist/amd/localstoragedb.js

```diff
@@ -0,0 +1,702 @@
+				dropTable(table_name);
+			},
+
+			// empty a table
+			truncate: function (table_name) {
+				tableExistsWarn(table_name);
+				truncate(table_name);
+			},
+
+			// alter a table
+			alterTable: function (table_name, new_fields, default_values) {
+				var result = false;
+				if (!validateName(table_name)) {
+					error("The database name '" + table_name + "' contains invalid characters");
+				} else {
+					if (typeof new_fields == "object") {
+						// make sure field names are valid
+						var is_valid = true;
+						for (var i = 0; i < new_fields.length; i++) {
+							if (!validateName(new_fields[i])) {
+								is_valid = false;
+								break;
+							}
+						}
+
+						if (is_valid) {
+							// cannot use indexOf due to <IE9 incompatibility
+							// de-duplicate the field list
+							var fields_literal = {};
+							for (var i = 0; i < new_fields.length; i++) {
+								fields_literal[new_fields[i]] = true;
+							}
+							delete fields_literal['ID']; // ID is a reserved field name
+
+							new_fields = [];
+							for (var field in fields_literal) {
+								if (fields_literal.hasOwnProperty(field)) {
+									new_fields.push(field);
+								}
+							}
+
+							alterTable(table_name, new_fields, default_values);
+							result = true;
+						} else {
+							error("One or more field names in the table definition contains invalid characters");
+						}
+					} else if (typeof new_fields == "string") {
+						if (validateName(new_fields)) {
+							var new_fields_array = [];
+							new_fields_array.push(new_fields);
+							alterTable(table_name, new_fields_array, default_values);
+							result = true;
+						} else {
+							error("One or more field names in the table definition contains invalid characters");
+						}
+					}
+				}
+
+				return result;
+			},
+
+			// number of rows in a table
+			rowCount: function (table_name) {
+				tableExistsWarn(table_name);
+				return rowCount(table_name);
+			},
+
+			// insert a row
+			insert: function (table_name, data) {
+				tableExistsWarn(table_name);
+				return insert(table_name, validateData(table_name, data));
+			},
+
+			// insert or update based on a given condition
+			insertOrUpdate: function (table_name, query, data) {
+				tableExistsWarn(table_name);
+
+				var result_ids = [];
+				if (!query) {
+					result_ids = getIDs(table_name);				// there is no query. applies to all records
+				} else if (typeof query == 'object') {				// the query has key-value pairs provided
+					result_ids = queryByValues(table_name, validFields(table_name, query));
+				} else if (typeof query == 'function') {				// the query has a conditional map function provided
+					result_ids = queryByFunction(table_name, query);
+				}
+
+				// no existing records matched, so insert a new row
+				if (result_ids.length == 0) {
+					return insert(table_name, validateData(table_name, data));
+				} else {
+					var ids = [];
+					for (var n = 0; n < result_ids.length; n++) {
```

Comment:
Missing #68 here.

I didn't get the idea of having localstoragedb.js duplicated.
Is it really necessary?